src/dist/replication/lib/replica_learn.cpp

```diff
@@ -749,10 +750,11 @@ void replica::on_copy_remote_state_completed(
             else
             {
                 derror(
-                    "%s: on_copy_remote_state_completed[%016llx]: learner = %s, learn duration = %" PRIu64 " ms, apply learned state from private log failed, err = %s, duration = %" PRIu64 " ns",
-                    name(), req.signature, req.learner.to_string(),
-                    err.to_string(),
+                    "%s: on_copy_remote_state_completed[%" PRIx64 "]: learner = %s, "
```

Comment:
ok

On Tue, Jun 21, 2016 at 10:41 AM, Zhenyu Guo <notifications@github.com>
wrote:

> In src/dist/replication/lib/replica_learn.cpp
> <https://github.com/imzhenyu/rDSN/pull/406#discussion_r67800214>:
>
> > @@ -749,10 +750,11 @@ void replica::on_copy_remote_state_completed(
> >              else
> >              {
> >                  derror(
> > -                    "%s: on_copy_remote_state_completed[%016llx]: learner = %s, learn duration = %" PRIu64 " ms, apply learned state from private log failed, err = %s, duration = %" PRIu64 " ns",
> > -                    name(), req.signature, req.learner.to_string(),
> > -                    err.to_string(),
> > +                    "%s: on_copy_remote_state_completed[%" PRIx64 "]: learner = %s, "
>
> I was trying to change all to PRIx64 to be more portable, but in the end I
> changed only this place to avoid more conflicts for the ongoing merge we
> have and plan to do that later after merge. Anyway, this change is mostly
> about the wrong order of the parameters in the "else" branch that lead to
> crash.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/406/files/45b188ed6f4b625ec572d3dc8d0bb2e490a079b2#r67800214>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgQPaS-kSDZ8toO7q7SqgAdNrehFCORks5qN09VgaJpZM4I6TiS>
> .
>